src/ts_catalog/chunk_data_node.c

```diff
@@ -233,13 +239,17 @@ List *
 ts_chunk_data_node_scan_by_chunk_id(int32 chunk_id, MemoryContext mctx)
 {
 	List *chunk_data_nodes = NIL;
-
+	ScanTupLock tuplock = {
+		.lockmode = LockTupleKeyShare,
```

Comment:
This isn't the right lock to take when doing `FOR UPDATE`.

Reply:
I also think we don't need to take a lock here in all cases, only when we read a tuple and maybe later want to delete it in the same TX (i.e., the FOR UPDATE case). Now we always grab the lock for reads.

Reply:
That was part which reads only, Exclusive is taken when we delete in functions below